src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/DeleteDocumentById.cs

```diff
@@ -13,27 +13,48 @@ namespace EdFi.DataManagementService.Backend.Postgresql.Operation;
 
 public interface IDeleteDocumentById
 {
-    public Task<DeleteResult> DeleteById(IDeleteRequest deleteRequest);
+    public Task<DeleteResult> DeleteById(
+        IDeleteRequest deleteRequest,
+        NpgsqlConnection connection,
+        NpgsqlTransaction transaction
+    );
 }
 
-public class DeleteDocumentById(
-    NpgsqlDataSource _dataSource,
-    ISqlAction _sqlAction,
-    ILogger<DeleteDocumentById> _logger
-) : IDeleteDocumentById
+public class DeleteDocumentById(ISqlAction _sqlAction, ILogger<DeleteDocumentById> _logger)
+    : IDeleteDocumentById
 {
-    public async Task<DeleteResult> DeleteById(IDeleteRequest deleteRequest)
+    private DeleteResult HandlePostgresException(string tableName, ITraceId traceId, PostgresException pe)
+    {
+        if (pe.SqlState == PostgresErrorCodes.SerializationFailure)
+        {
+            _logger.LogDebug(
+                pe,
+                "Transaction conflict on {TableName} table delete - {TraceId}",
+                tableName,
+                traceId
+            );
+            return new DeleteResult.DeleteFailureWriteConflict();
```

Comment:
Is this just a failure scenario for "first in, wins"? Client application needs to decide on its next steps? What status code will we end up with?

Reply:
Yes, this is concurrent modification detection which results in a 409 Conflict back to the client.  These are the "your transaction deleted it at the same time another client transaction updated/deleted it, and they finished first" type errors.

The typical client action would be to either 1) be concerned someone else is mucking with their stuff, or 2) not care and just retry. If a few retries fail, DMS is probably overloaded and the client needs to handle it similar to a rate limit scenario.

As an aside, datastores like MongoDB typically default to auto retrying a few times before reporting a conflict, since unless there is a very heavy DB load a retry is sufficient.